src/rideManager.ts

```diff
@@ -0,0 +1,184 @@
+import { Group, Ride } from '../typings/ride'
+import Bot from 'node-telegram-bot-api'
+import * as db from './database.js'
+import { weekdays, emojis } from '../utils/const.js'
+import * as format from '../utils/format.js'
+import { getUserLink } from '../utils/bot.js'
+
+export default class RideManager {
+  constructor() {
+    db.connectToDatabase()
+  }
+
+  public async addRide(
+    chatId: number,
+    rideInfo: { user: Bot.User; time: Date; description: String; direction: string }
+  ): Promise<boolean> {
+    const ride = {
+      full: 0,
+      ...rideInfo
+    }
+
+    return await db.updateGroup(
+      chatId,
+      {
+        $set: {
+          [rideInfo.direction + '.' + rideInfo.user.id]: ride
+        }
+      },
+      { upsert: true }
+    )
+  }
+
+  public async removeRide(
+    chatId: number,
+    rideInfo: { userId: number; direction: string }
+  ): Promise<boolean> {
+    return await db.updateGroup(
+      chatId,
+      {
+        $unset: {
+          [rideInfo.direction + '.' + rideInfo.userId]: ''
+        }
+      },
+      { upsert: false }
+    )
+  }
+
+  public async setRideFull(
+    chatId: number,
+    rideInfo: { userId: number; direction: string; state: number }
+  ): Promise<boolean> {
+    return await db.updateGroup(
+      chatId,
+      {
+        $set: {
+          [rideInfo.direction + '.' + rideInfo.userId + '.full']: rideInfo.state
+        }
+      },
+      { upsert: false }
+    )
+  }
+
+  public async cleanRides(chatId: number, now: Date) {
+    const docs = (await db.scrapeGroupRides(chatId)) as Group
```

Comment:
It could, but I wasn't sure of assuming that the type will be that one, thinking about the scenario where we don't have any docs being retrieved in the query